custom_models/import_models/llama-3/llama3-sftt-llama3-fine-tuning.ipynb

```diff
@@ -0,0 +1,664 @@
+    "\n",
+    "We will quantize the model as bf16 model. We use [Supervised Fine-tuning Trainer](https://huggingface.co/docs/trl/sft_trainer) (SFT) for fine tuning the model. We will use Anthropic/Vicuna like Chat Template with User: and Assistant: roles to fine tune the model. We will use [HuggingFaceH4/no_robots] (https://huggingface.co/datasets/HuggingFaceH4/no_robots) dataset for fine tuning the model. This is a high-quality dataset of 10,000 instructions and demonstrations created by skilled human annotators. Using FSDP and Q-Lora allows us to fine tune Llama-3 models on 2x consumer GPU's. For fine tuning in this notebook we use ml.g5.12xlarge as a SageMaker Training Job.\n",
+    "\n",
+    "[Amazon SageMaker](https://aws.amazon.com/sagemaker) provides a fully managed service that enables build, train and deploy ML models at scale using tools like notebooks, debuggers, profilers, pipelines, MLOps, and more – all in one integrated development environment (IDE). [SageMaker Model Training](https://aws.amazon.com/sagemaker/train/) reduces the time and cost to train and tune machine learning (ML) models at scale without the need to manage infrastructure.\n",
+    "\n",
+    "This notebook is inspired by Philipp Schmid Blog - https://www.philschmid.de/fsdp-qlora-llama3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5da662ec",
+   "metadata": {},
+   "source": [
+    "## Model License information"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2541e27d",
+   "metadata": {},
+   "source": [
+    "In this notebook we use the Meta Llama3 model from HuggingFace. This model is a gated model within HuggingFace repository. To use this model you have to agree to the license agreement (https://llama.meta.com/llama3/license) and request access to the model before it can be used in this notebook."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c2860a3-68b3-435d-8d2c-753c90a24da3",
+   "metadata": {},
+   "source": [
+    "### Install the Pre-Requisites"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "885a83fd-92b8-47ad-9128-c820fb4a6188",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install transformers \"sagemaker>=2.190.0\" \"huggingface_hub\" \"datasets[s3]==2.18.0\" --upgrade --quiet\n",
+    "!pip install boto3 s3fs \"aiobotocore==2.11.0\" --upgrade --quiet"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "986ce221-9721-4089-8f5f-9c2d4afba925",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from huggingface_hub import notebook_login\n",
+    "notebook_login()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe5f5960-12b5-4310-8711-323dde090b96",
+   "metadata": {},
+   "source": [
+    "### Setup\n",
+    "We will initialize the SageMaker Session required to finetune the model."
```

Comment:
add links to sagemaker training

Reply:
Included link to https://aws.amazon.com/sagemaker/train/